lib/application.js

```diff
@@ -121,6 +136,105 @@ class Application extends api.events.EventEmitter {
     });
   }
 
+  loadLib(path, callback) {
+    api.fs.readdir(path, (err, files) => {
+      if (err) {
+        impress.log.error(impress.CANT_READ_DIR + path);
+        callback(err);
+        return;
+      }
+      api.metasync.each(files, (fileName, cb) => {
+        const filePath = api.path.join(path, fileName);
+        api.fs.stat(filePath, (err, stats) => {
+          if (err) {
+            impress.log.error(impress.CANT_READ_FILE + filePath);
+            cb();
+            return;
+          }
+          if (stats.isDirectory()) {
+            this.loadLib(filePath, cb);
+          } else {
+            this.loadLibFile(filePath, cb);
+          }
+        });
+      }, (err) => {
+        this.cache.watch(api.path.relative(this.dir, path));
+        callback(err);
+      });
+    });
+  }
+
+  loadLibFile(path, callback) {
+    if (!path.endsWith('.js')) {
+      callback(null);
+      return;
+    }
+
+    const isTest = path.endsWith('.test.js');
+    if (isTest && impress.mode !== 'test') {
+      callback(null);
+      return;
+    }
+
+    impress.createScript(this, path, (err, test) => {
+      if (err) {
+        callback(err);
+        return;
+      }
+      if (isTest) this.tests.unit.push({ path, test });
+      callback(null);
+    });
+  }
+
+  loadIntegrationTests(path, callback) {
+    if (!path) {
+      path = api.path.join(this.dir, 'test');
+    }
+    api.fs.readdir(path, (err, files) => {
+      if (err) {
+        impress.log.error(impress.CANT_READ_DIR + path);
+        callback(err);
+        return;
+      }
+      api.metasync.each(files, (fileName, cb) => {
+        const filePath = api.path.join(path, fileName);
+        api.fs.stat(filePath, (err, stats) => {
+          if (err) {
+            impress.log.error(impress.CANT_READ_FILE + filePath);
+            cb();
+            return;
+          }
+          if (stats.isDirectory()) {
+            this.loadIntegrationTests(filePath, cb);
+          } else {
+            this.loadIntegrationTestFile(filePath, cb);
+          }
+        });
+      }, callback);
+    });
+  }
+
+  loadIntegrationTestFile(path, callback) {
+    if (impress.mode !== 'test') {
```

Comment:
Is this check needed? You are checking the `mode` before calling `loadIntegrationTests`.